### PR TITLE
Allow using VSphereClusterIdentity based identityRef

### PR DESCRIPTION
**Pull request by DanielXiao**
API consumers like TMC can not save any vCenter credentials, so it can not create a Secret based identityRef for each Cluster. It relies on TKG admin to create a VSphereClusterIdentity shared by multiple Clusters.

In this patch a Cluster variable is introduced to allow users to specify an alternative identityRef. Nevertheless, Tanzu cli still use Secret as the default identity type to keep backward UX compatible.

This patch also enhances VM template resolver to get vCenter credentials from any types of identityRef.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by DanielXiao** [comment]
Wait for https://github.com/vmware-tanzu/tanzu-framework/pull/3849 to be merged.

**Review by tenczar — APPROVED** [review]
This looks good to me. Have we done any tests with a VSphereClusterIdentity that specifies allowedNamespaces? I want to make sure that the cli can't leak these credentials to namespaces that aren't allowed to use them. In particular a workload cluster should not be created if it references a VSphereClusterIdentity that doesn't have the namespace of the workload cluster in its allowedNamespaces list. If we have validated that, then this looks good to me.

**Comment by DanielXiao** [comment]
Good point. I will add checks if the cluster meet allowedNamespaces.

File: packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-service-account.yaml [hunk 1]
```diff
@@ -29,41 +29,6 @@ rules:
       - get
       - list
       - delete
-      - watch
```
Comment (wjun) [inline packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-service-account.yaml:32]: Why these rules are not required?

Comment (DanielXiao) [reply packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-service-account.yaml:32]: It was introduced by https://github.com/vmware-tanzu/tanzu-framework/pull/3683 and should be cleaned up by https://github.com/vmware-tanzu/tanzu-framework/pull/3849.

**Review by wjun — APPROVED** [review]

**Pull request merged**